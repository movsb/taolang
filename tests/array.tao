function main() {
    let a = [1,true,nil,"str",{},1+2*3,];
    println(a);
    println(a.length);
    println(a[3]);
    println([9][0]);

    {
        let a = [1,3,5,7,9];
        let n = a.unshift(-1,-2);
        println(a, n);
    }

    {
        let myFish = ["angel", "clown", "mandarin", "sturgeon"];
        let removed = myFish.splice(2, 0, "drum");
        println(myFish);
        println(removed);
    }
    {
        let myFish = ["angel", "clown", "drum", "mandarin", "sturgeon"];
        let removed = myFish.splice(3, 1);
        println(myFish);
        println(removed);
    }
    {
        let myFish = ["angel", "clown", "drum", "sturgeon"];
        let removed = myFish.splice(2, 1, "trumpet");
        println(myFish);
        println(removed);
    }
    {
        let myFish = ["angel", "clown", "trumpet", "sturgeon"];
        let removed = myFish.splice(0, 2, "parrot", "anemone", "blue");
        println(myFish);
        println(removed);
    }
    {
        let myFish = ["parrot", "anemone", "blue", "trumpet", "sturgeon"];
        let removed = myFish.splice(myFish.length - 3, 2);
        println(myFish);
        println(removed);
    }
    {
        let myFish = ["angel", "clown", "mandarin", "sturgeon"];
        let removed = myFish.splice(-2, 1);
        println(myFish);
        println(removed);
    }
    {
        let myFish = ["angel", "clown", "mandarin", "sturgeon"];
        let removed = myFish.splice(2);
        println(myFish);
        println(removed);
    }
    {
        let a = [1,2,3];
        a.push(4);
        a.push("5");
        a.push([1,2,3]);
        a.push({a:8});
        a.push(1,true,[],{},"s");
        println(a);
    }
    {
        let a = [1,2,3];
        println(a.pop());
        println(a.pop());
        println(a.pop());
        println(a.pop());
    }
    {
        println([1,2,3].join());
        println([1,2,3].join(""));
        println([1,2,3].join(true));
        println([1,2,3].join(nil));
        println([1,2,3].join({},[]));
    }
}
