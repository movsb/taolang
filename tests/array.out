[1,true,nil,str,{},7]
6
str
9
[-2,-1,1,3,5,7,9] 7
[angel,clown,drum,mandarin,sturgeon]
[]
[angel,clown,drum,sturgeon]
[mandarin]
[angel,clown,trumpet,sturgeon]
[drum]
[parrot,anemone,blue,trumpet,sturgeon]
[angel,clown]
[parrot,anemone,sturgeon]
[blue,trumpet]
[angel,clown,sturgeon]
[mandarin]
[angel,clown]
[mandarin,sturgeon]
